Sources/AsyncHTTPClient/ConnectionPool.swift

```diff
@@ -103,14 +103,19 @@ final class ConnectionPool {
 
     func close(on eventLoop: EventLoop) -> EventLoopFuture<Bool> {
         let providers = self.lock.withLock {
-            self.providers.values
+            // we materialize Values into an Array before returning from under a lock
+            // as thread safety analyzer considers safety of Values as undefined and dependant on implementation
+            // e.g. a lazy collection implementation could by thread unsafe to be returned from within a lock
+            Array(self.providers.values)
```

Comment:
I do not believe this fix is necessary: I think TSAN is wrong.

Reply:
I am lacking deep knowledge of Swift runtime / libraries here :(.

Is it possible that @ktoso could advise here?

I seen in person issues with returning a lazy collection out of lock guarded block in Scala many years ago and performing of "materialise" was a go-to tool then.

Reply:
While the dictionary values view is "lazy", conceptually, it is nonetheless still a value type with value semantics. That means that changes to the dictionary to which it points must not manifest in changes to this value. _Acquiring_ the view must be done with a lock, but once it's copied out we're safe. Either TSAN is wrong, or Dictionary in Swift 5.3 is implemented incorrectly.

Reply:
PR is updated to contain only `.count` fix according to the discussion above.